src/content/app/entity-viewer/variant-view/transcript-consequences/TranscriptConsequences.tsx

```diff
@@ -52,6 +61,39 @@ const TranscriptConsequences = (props: Props) => {
     variantId,
     alleleId: activeAlleleId
   });
+  const dispatch = useAppDispatch();
+
+  useExpandedTranscriptConsequence({
+    genomeId: genomeId,
+    variantId: variantId,
+    alleleId: activeAlleleId,
+    expandIds: []
+  });
+
+  const transcriptConseqeuencesToExpandByDefault: string[] = [];
+  const expandedTranscriptIds = useAppSelector((state) =>
+    getExpandedTranscriptConseqeuenceIds(
+      state,
+      genomeId,
+      variantId,
+      activeAlleleId
+    )
+  );
+
+  useEffect(() => {
+    // expandedTranscriptIds will be null only on the initial load, before user interaction.
+    // Expand default transcript ids only when expandedTranscriptIds.
+    if (expandedTranscriptIds === null) {
+      dispatch(
+        toggleTranscriptConsequenceIds(
+          genomeId,
+          variantId,
+          activeAlleleId,
+          transcriptConseqeuencesToExpandByDefault
+        )
+      );
+    }
+  }, [transcriptConseqeuencesToExpandByDefault, expandedTranscriptIds]);
```

Comment:
Why do you have both this hook, and the `useExpandedTranscriptConsequence` hook? Is the `useExpandedTranscriptConsequence` hook needed anymore?

Reply:
When the `useExpandedTranscriptConsequence` is called first, it initialises the state. And later when we get the transcripts to expand after sorting and ordering, this hook gets called.